model/parse_cloudlab.py

```diff
@@ -0,0 +1,644 @@
+                    # We do not need to check whether recv_time_prev
+                    # is -1 (unknown) because the windowed metrics
+                    # skip the case where j == 0.
+                    win_state[win]["loss_queue_estimate"], new = loss_rate(
+                        win_state[win]["loss_queue_estimate"], win_start_idx,
+                        pkt_loss_cur_estimate, recv_time_cur, recv_time_prev,
+                        win_size_us, j)
+                elif "loss rate true" in metric:
+                    # We do not need to check whether recv_time_prev
+                    # is -1 (unknown) because the windowed metrics
+                    # skip the case where j == 0.
+                    win_state[win]["loss_queue_true"], new = loss_rate(
+                        win_state[win]["loss_queue_true"], win_start_idx,
+                        pkt_loss_cur_true, recv_time_cur, recv_time_prev,
+                        win_size_us, j)
+                elif "queue occupancy" in metric:
+                    # Queue occupancy is calculated using the router
+                    # logs, below.
+                    continue
+                elif "mathis model throughput p/s" in metric:
+                    # Use the loss event rate to compute the Mathis
+                    # model fair throughput.
+                    loss_rate_estimate = (
+                        pkt_loss_total_estimate / j if j > 0 else -1)
+                    new = utils.safe_div(
+                        MATHIS_C,
+                        utils.safe_div(
+                            utils.safe_mul(
+                                min_rtt_us,
+                                utils.safe_sqrt(
+                                    loss_rate_estimate)),
+                            1e6))
+                elif "mathis model label" in metric:
+                    # Use the current throughput and Mathis model
+                    # fair throughput to compute the Mathis model
+                    # label.
+                    new = utils.safe_mathis_label(
+                        output[j][make_win_metric(
+                            "average throughput p/s", win)],
+                        output[j][make_win_metric(
+                            "mathis model throughput p/s", win)])
+                else:
+                    raise Exception(f"Unknown windowed metric: {metric}")
+                output[j][metric] = new
+        unfair_flws.append(output)
+
+    # Save memory by explicitly deleting the sent and received packets
+    # after they have been parsed. This happens outside of the above
+    # for-loop because only the last iteration's sent and received
+    # packets are not automatically cleaned up by now (they go out of
+    # scope when the sent_pkts and recv_pkts variables are overwritten
+    # by the next loop).
+    del sent_pkts
+    del recv_pkts
+    del ack_pkts
+
+    # Determine if there are any NaNs or Infs in the results. For the
```

Comment:
Skipping all the queue_occupancy features since we don't have the pcap file from bottleneck router.